### PR TITLE
Issue #1152 Replace Coordination Service with new ReadWriteLock Impl

### PR DESCRIPTION
**Pull request by chrisdennis**

### CODE REVIEW

File: clustered/client/src/main/java/org/ehcache/clustered/lock/client/VoltronReadWriteLock.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import org.terracotta.connection.Connection;
+import org.terracotta.connection.entity.EntityRef;
+import org.terracotta.exception.EntityAlreadyExistsException;
+import org.terracotta.exception.EntityNotFoundException;
+import org.terracotta.exception.EntityNotProvidedException;
+import org.terracotta.exception.EntityVersionMismatchException;
+
+/**
+ *
+ * @author cdennis
```
Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/lock/client/VoltronReadWriteLock.java]: Code style: no author tags

File: clustered/client/src/main/java/org/ehcache/clustered/lock/client/VoltronReadWriteLockClient.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    });
+  }
+
+  @Override
+  public void close() {
+    endpoint.close();
+  }
+
+  public boolean tryLock(HoldType type) {
+    while (true) {
```
Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/lock/client/VoltronReadWriteLockClient.java]: Never loops

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntityFactory.java [hunk 3]
```diff
@@ -91,58 +79,72 @@ public void abandonLeadership(String entityIdentifier) {
...
+    Hold existingMaintenance = maintenanceHolds.get(identifier);
+    Hold localMaintenance = null;
+    if (existingMaintenance == null) {
+      localMaintenance = createAccessLockFor(identifier).tryWriteLock();
+    }
+    if (existingMaintenance == null && localMaintenance == null) {
+      String message = "Unable to create entity for cluster id " + identifier
+          + ": another client owns the maintenance lease";
+      LOGGER.error(message);
+      throw new EhcacheEntityCreationException(message);
```
Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntityFactory.java]: Seems weird to both log and throw - why is throw not enough?

Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntityFactory.java]: Especially as the code path in `DefaultClusteringService.start` will most likely print this message when `auto-create` is enabled and the entity already exists and is in use.

**Comment by chrisdennis** [comment]
@ljacomet corrections applied

**Pull request merged**